common:
    lid_topic:  "/rslidar_rs80_points"
    imu_topic:  "/imu_raw"
    time_sync_en: false         # ONLY turn on when external time synchronization is really not possible
    # Frames
    lidarFrame: "base_link"                     # 
    baselinkFrame: "base_link"                  # 
    odometryFrame: "map"                       # 
    mapFrame: "map"

preprocess:
    lidar_type: 6                # 1 for Livox serials LiDAR, 2 for Velodyne LiDAR, 3 for ouster LiDAR, 4 for ls 16,  5 for RSHelios32
    scan_line: 80
    scan_rate: 10                # only need to be set for velodyne, unit: Hz,
    timestamp_unit: 0            # the unit of time/t field in the PointCloud2 rostopic: 0-second, 1-milisecond, 2-microsecond, 3-nanosecond.
    blind: 4                     # 盲区
    blind_far: 150
    lidar_rot: [0,0,0]           # lidar rotation rpy
    cut_frame_num: 0             # must be positive integer
    

mapping:
    acc_cov: 0.1
    gyr_cov: 0.1
    b_acc_cov: 0.0001
    b_gyr_cov: 0.0001
    fov_degree:    180
    det_range:     150.0
    extrinsic_est_en:  false      # true: enable the online estimation of IMU-LiDAR extrinsic,
    extrinsic_T: [ 0, 0, 0]
    extrinsic_R: [ 1, 0, 0, 
                   0, 1, 0, 
                   0, 0, 1]

publish:
    path_en:  true
    scan_publish_en:  true       # false: close all the point cloud output
    dense_publish_en: true       # false: low down the points number in a global-frame point clouds scan.
    scan_bodyframe_pub_en: true  # true: output the point cloud scans in IMU-body-frame

pcd_save:
    pcd_save_en: false
    interval: -1                 # how many LiDAR frames saved in each pcd file; 
                                 # -1 : all frames will be saved in ONE pcd file, may lead to memory crash when having too much frames.
imu:
  # imu rpy weight
  imuRPYEnable: true
  imuRPYWeight: [0.8, 0.8, 0] 
  angleDiff: 0.0017

feature_extract_enable: 0
point_filter_num: 4 # 对当前点云的filter
max_iteration: 3 #最大迭代次数
filter_size_surf: 0.5 #对当前点云的filter
filter_size_map: 0.5 #对地图的filter
cube_side_length: 1000
runtime_pos_log_enable: 0